viziphant/array_data.py

```diff
@@ -0,0 +1,163 @@
+        i-th channel in the AnalogSignal object, False will plot a black
+        square. If None, all electrodes are plotted.
+        Default: None
+    color_map : matplotlob.pyplot.cm.get_cmap
+        Colormap to for plotting the data. If None, the current color map is
+        used. 
+        Default: None
+    arrow_scale_factor : float
+        Factor by which to scale the direction arrows in each square.
+        Default: 1
+    summary_arrow_scale_factor : float
+        Factor by which to scale the summary direction arrows in the four array
+        quadrants. If None, these arrows are not plotted.
+        Default: None
+    rotation : float
+        Rotates the array CW by `rotation` degree. Currently partly supported,
+        directions are not yet rotated. Default: 0.
+
+
+    Returns
+    =======
+    numpy.array
+        Image returned by pcolormesh.
+        
+    Example
+    =======
+    # Create fake data for a 4x25 electrode array, that is the array is 4 
+    # electrodes high and 25 wide. Electrode 1 is bottom left, electrode 100 
+    # is top right. We set channel to a high value. Also, for simplicity, we'll
+    # create only one sample of the AnalogSignal.
+    >>> grid_data = np.random.random((1,100))
+    >>> grid_data[0,3] = 3
+    >>> grid_data_obj = neo.AnalogSignal(
+            grid_data*pq.mV, t_start=0*pq.s, sampling_rate=1000*pq.Hz)
+    
+    # Create corresponding directions in the direction 0 rad
+    direction_data = np.exp(np.complex(0,1)*np.random.random((1,100))*.5)
+    >>> direction_data_obj = neo.AnalogSignal(
+            direction_data*pq.dimensionless, 
+            t_start=0*pq.s, sampling_rate=1000*pq.Hz)
+    >>> cmap = plt.cm.get_cmap('hsv')
+    
+    # Plot the data
+    >>> plot_array(
+            plt.gca(), (4,25), grid_data_obj, direction_data_obj, 
+            arrow_scale_factor=.7, summary_arrow_scale_factor=.5, 
+            color_map=cmap)
+    >>> plt.show()
+    """
+    if not color_map:
+        color_map = matplotlob.pyplot.cm.get_cmap()
+        
+    # Broken electrodes
+    color_map.set_bad(color='k')
+
+    # If no electrode mask is given, select all electrodes
+    if not electrode_mask:
+        electrode_mask = [True] * (dimension[0]*dimension[1])
+    # Arrange data on a grid
+    grid_data = np.ma.array(np.reshape(
+        array_data, dimension), mask=(np.equal(electrode_mask, False)))
+
+    # Plot phases as pcolormesh
+    X, Y = np.meshgrid(
+        np.arange(0, dimension[1]+1), np.arange(0, dimension[0]+1))
+
+    if rotation != 0:
+        # rotate clockwise
+        rotation = 2.*np.pi-rotation
+
+        # affine transform
+        X = X - (dimension[0]/2-0.5) #4.5
+        Y = Y - (dimension[1]/2-0.5) #4.5
+        Xr = X * np.cos(rotation) - Y * np.sin(rotation)
+        Yr = X * np.sin(rotation) + Y * np.cos(rotation)
+        X = Xr + (dimension[0]/2-0.5)
+        Y = Yr + (dimension[0]/2-0.5)
```

Comment:
Should this be a 'dimension[1]' instead of 'dimension[0]'?
```suggestion
        Y = Yr + (dimension[1]/2-0.5)
```

Reply:
Good catch, however, actually it must be 1 for X and 0 for Y here and further up....